core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -58,11 +63,148 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
 
   @Before
   override def setUp() {
-    super.setUp
+    super.setUp()
     client = AdminClient.createSimplePlaintext(this.brokerList)
     TestUtils.createTopic(this.zkUtils, topic, 2, serverCount, this.servers)
   }
 
+  @After
+  override def tearDown() {
+    client.close()
+    super.tearDown()
+  }
+
+  @Test
+  def testPurgeDataBeforeWithSeekToBeginning() {
+    val consumer = consumers.head
+
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
```

Comment:
It seems that we don't have to worry about CPU burn because `TestUtils.waitUntilTrue()` will wait for 100L between calls to `poll(0)`.

Sure. I updated the patch to move this wait logic into a separate method.